### PR TITLE
Generate code for Bigtable v2 protobufs

### PR DESCRIPTION
**Pull request by tseaver**
Put V2-generated files them in a separate `_generated_v2` subdir, to ease migration.

Attempt to factor generation process for clarity (toward #1482).

Supersedes #1895.

### CODE REVIEW

**Comment by daspecster** [comment]
@tseaver & @dhermes is there anything specific I should keep an eye out for with these?

**Comment by tseaver** [comment]
@daspecster The code in the `_generated_v2` package you can ignore.  The other changes are attempts to improve how we generated code from protobufs (splitting up the Makefile, etc.)

**Comment by daspecster** [comment]
Should datastore be in here with bigtable or should that be a separate PR?

**Comment by tseaver** [comment]
The only thing I did with `datastore` was to factor its generate-from-protobuf bits into a separate `Makefile.datastore`.  I ran the generation for it, and saw minor changes to the generated code, which I then discarded (not on my radar just now).

File: Makefile.bigtable_v1 [hunk 1]
```diff
@@ -15,8 +16,8 @@ help:
 
 generate:
 	# Retrieve git repos that have our *.proto files.
-	[ -d cloud-bigtable-client ] || git clone https://github.com/GoogleCloudPlatform/cloud-bigtable-client --depth=1
-	cd cloud-bigtable-client && git pull origin master
+	[ -d $(BIGTABLE_CHECKOUT_DIR) ] || git clone https://github.com/GoogleCloudPlatform/cloud-bigtable-client --depth=1
+	cd $(BIGTABLE_CHECKOUT_DIR) && git pull origin master
```
Comment (daspecster) [inline Makefile.bigtable_v1]: Does this work?

Example...
```bash
$ export MYDIR=/wewefw/wefwe/
$ cd $(MYDIR)
zsh: command not found: MYDIR
```

**Comment by daspecster** [comment]
LGTM!

**Pull request merged**